pdns/dnsdist-console.cc

```diff
@@ -45,6 +46,25 @@ void feedConfigDelta(const std::string& line)
   g_confDelta.push_back({now,line});
 }
 
+string historyFile(const bool &ignoreHOME = false)
+{
+  string ret;
+
+  struct passwd pwd;
+  struct passwd *result;
+  char buf[16384];
+  getpwuid_r(getuid(), &pwd, buf, sizeof(buf), &result);
+
+  const char *homedir = getenv("HOME");
+  ret = "."; // CWD if nothing works..
+  if (result)
+    ret = string(pwd.pw_dir);
+  if (homedir && !ignoreHOME) // $HOME overrides what the OS tells us
+    ret = string(homedir);
+  ret.append("/.dnsdist_history");
```

Comment:
How about not setting `ret` to `"."` beforehand and testing here if it's empty? If it is (somehow `$HOME` or `pwd.pw_dir` were empty or nothing worked), we could then fallback to `"."` before appending `"/.dnsdist_history"`. This would prevent trying to write in `"/.dnsdist_history"`.